core/domain/change_domain.py

```diff
@@ -74,6 +74,15 @@ def validate_cmd(cmd_name, valid_cmd_attribute_specs, actual_cmd_attributes):
     if error_msg_list:
         raise utils.ValidationError((', ').join(error_msg_list))
 
+    deprecated_values = valid_cmd_attribute_specs.get('deprecated_values')
```

Comment:
Should this use [] accessor instead so that it throws when the key isn't present? Better to be strict, in general.

Reply:
The deprecated_values key is optional like the allowed_values. So the exception should not be thrown if the key is not present.